platform/windows/detect.py

```diff
@@ -365,15 +383,12 @@ def configure_mingw(env):
     if os.name != "nt":
         env["PROGSUFFIX"] = env["PROGSUFFIX"] + ".exe"  # for linux cross-compilation
 
-    if env["bits"] == "default":
-        if os.name == "nt":
-            env["bits"] = "64" if "PROGRAMFILES(X86)" in os.environ else "32"
-        else:  # default to 64-bit on Linux
-            env["bits"] = "64"
-
     mingw_prefix = ""
 
-    if env["bits"] == "32":
+    # TODO: This doesn't seem to be working, or maybe I just have
+    # MinGW set up incorrectly. It always gives me x86_64 builds,
+    # even though arch == "x86_32" and the file name has x86_32 in it.
```

Comment:
Depending on setup, MinGW can have either prefixed binaries (e.g. `x86_64-w64-mingw32-gcc`) or everything in the subfolder (`gcc`) in this case which one used is determined by `path`. I'm not sure how to detect arch in this case, we can do something like `detect_visual_c_compiler_version`, but MinGW path is not standard.